iocBoot/templates/st.cmd.template

```diff
@@ -1,10 +1,51 @@
 #!{{hashbang}}
-###### AUTO-GENERATED DO NOT EDIT ##############
+################### AUTO-GENERATED DO NOT EDIT ###################
+{% set project, plc = get_plc_by_name(projects, plc) -%}
+{%- set plc_host = plc_hostname_override or plc.target_ip %}
+#
+#         Project: {{ project.filename.name }}
+#        PLC name: {{ plc.name }} ({{ plc.link_name }})
+# Generated using: pytmc {{ pytmc_version}}
+# Project version: {{ project.git_info.describe }}
+#    Project hash: {{ project.git_info.sha }}
+#     PLC IP/host: {{ plc.target_ip}} ({{ plc_hostname_override }})
+#
+# Libraries:
+#
+{% for library, info in get_library_versions(plc) | dictsort %}
+{% if "PlaceholderReference" in info and "PlaceholderResolution" in info %}
+#   {{ library }}: {{ info.PlaceholderReference.version }} -> {{ info.PlaceholderResolution.version }} ({{ info.PlaceholderResolution.vendor }})
+{% elif "PlaceholderReference" in info %}
+#   {{ library }}: {{ info.PlaceholderReference.version }} ({{ info.PlaceholderReference.vendor }})
+{% endif %}
+{% endfor %}
+#
+################### AUTO-GENERATED DO NOT EDIT ###################
+{%- if plc.tmc %}
+
+{%- set allow_errors = ignore_linter_errors | int %}
+{%- set db_filenames, records = generate_records(plc, allow_errors=allow_errors) %}
+{%- set num_records = records | length %}
+{%- set max_params = num_records + 1000 %}
+
+{%- else %}
+
+{%- set db_filenames = [] %}
+{%- set records = {} %}
+{%- set num_records = 0 %}
+{%- set max_params = 10000 %}
+
+{%- endif %}
 
 < envPaths
 
 epicsEnvSet("ADS_IOC_TOP", "$(TOP)" )
 
+{% if add_plc_route | int != 0 %}
+# Add a route to the PLC automatically:
+system("${ADS_IOC_TOP}/scripts/add_route.sh {{ plc_host }} {{ ioc_host_ip_regex }}")
```

Comment:
I'm excited to see how well this works. I suspect the only time it fails is when the IOC launches prior to the PLC launch?

One question: where exactly is the `ioc_host_ip_regex` set up? I see the Makefile variable (in ALL CAPS) but nothing else.

Reply:
>  I suspect the only time it fails is when the IOC launches prior to the PLC launch?

Yes, but the IOC will also restart if it can't communicate with the PLC, so it should work out 🤞 

> One question: where exactly is the ioc_host_ip_regex set up? I see the Makefile variable (in ALL CAPS) but nothing else.

It's macro-ified here, along with all of the rest of the ads-ioc-specific variables:
https://github.com/klauer/ads-ioc/blob/1bf04d99caf4661ae9ebdedd6961a4242109bda1/iocBoot/templates/Makefile.base#L96